fairing/deployers/serving/serving.py

```diff
@@ -76,13 +80,13 @@ def generate_service_spec(self):
                     name="serving",
                     port=5000
                 )],
-                type="LoadBalancer",
+                type="ClusterIP",
```

Comment:
ClusterIP is only routable from inside the cluster. In Kubeflow notebooks, thats ok. For local + other notebooks, this won't work.

Reply:
For outside the cluster can we reference the ingress for the cluster which should be secured?

I filed some issues for this:
#208 - Add ISTIO routing rules
#200 - Add auth to prediction requests 

So with GCP for example we'd print the full URL
https://${APP}.endpoints.${PROJECT}.cloud.goog

And then attach a JWT for IAP to the prediction request.

Reply:
I see how it can be made to point to the IAP endpoint on GCP, but thats not the change proposed. And how would this work for generic use cases?

Changing to clusterIP without making those other changes would break all users

Reply:
I made it an argument with a default of ClusterIP. Does that work?

Folks depending on LoadBalancer will need to update the code. Do we have examples that would need to be updated. 

We could leave the default as type LoadBalancer but I'd prefer not to be "open" by default. Especially since the target audience (data scientists) is unlikely to understand the risks of doing so.

ClusterIp will work for in cluster traffic and for port-forwarding. 

For external traffic I was thinking users would do one of the following
  1. Create a suitable ingress pointing at the Kubeflow ISTIO gateway (akin to IAP)
  1. Chose to expose the service either via type LoadBalancer or via Ingress

If folks want to do #2 I think its fine I just think they should have to explicitly choose to do so.